File: extract_css.py
import re
import os

html_path = "ClonAfish/Афиша Краснодара 2025-2026 - куда сходить в Краснодаре - мероприятия и события на сегодня, завтра, выходные _ 😋 KASSIR.RU.html"
css_dest = "web/app/original.css"

try:
    with open(html_path, "r", encoding="utf-8") as f:
        content = f.read()

    # Find the massive style block. It starts with <style>@font-face
    start_marker = "<style>@font-face"
    end_marker = "</style>"
    start_idx = content.find(start_marker)

    if start_idx != -1:
        # Find the end of THIS style tag
        end_idx = content.find(end_marker, start_idx)
        if end_idx != -1:
            css_content = content[start_idx + len("<style>"):end_idx]
            with open(css_dest, "w", encoding="utf-8") as f:
                f.write(css_content.strip())
            print(f"CSS extracted, size: {len(css_content)} bytes")
        else:
            print("End style tag not found")
    else:
        print("Start style tag not found")
except Exception as e:
    print(f"Error extracting CSS: {e}")

# Check asset dir count
asset_dest = "web/public/original"
try:
    if os.path.exists(asset_dest):
        files = []
        for r, d, f in os.walk(asset_dest):
            for file in f:
                files.append(os.path.join(r, file))
        print(f"Assets in {asset_dest}: {len(files)} files")
    else:
        print("Asset dest not found")
except Exception as e:
    print(f"Error checking assets: {e}")

File: extract_css.js
const fs = require('fs');
const path = require('path');

const htmlPath = "ClonAfish/Афиша Краснодара 2025-2026 - куда сходить в Краснодаре - мероприятия и события на сегодня, завтра, выходные _ 😋 KASSIR.RU.html";
const cssDest = "web/app/original.css";

try {
    const content = fs.readFileSync(htmlPath, 'utf-8');
    const startMarker = "<style>@font-face";
    const endMarker = "</style>";
    const startIdx = content.indexOf(startMarker);

    if (startIdx !== -1) {
        const endIdx = content.indexOf(endMarker, startIdx);
        if (endIdx !== -1) {
            // Find ALL content inside this style tag
            const cssContent = content.substring(startIdx + "<style>".length, endIdx);
            fs.writeFileSync(cssDest, cssContent.trim(), 'utf-8');
            console.log(`CSS extracted, size: ${cssContent.length} bytes`);
        } else {
            console.log("End style tag not found");
        }
    } else {
        console.log("Start style tag not found");
    }
} catch (e) {
    console.error("Error:", e);
}

// Check assets
const assetDest = "web/public/original";
try {
    const getAllFiles = function (dirPath, arrayOfFiles) {
        files = fs.readdirSync(dirPath)
        arrayOfFiles = arrayOfFiles || []
        files.forEach(function (file) {
            if (fs.statSync(dirPath + "/" + file).isDirectory()) {
                arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles)
            } else {
                arrayOfFiles.push(path.join(dirPath, "/", file))
            }
        })
        return arrayOfFiles
    }
    if (fs.existsSync(assetDest)) {
        const files = getAllFiles(assetDest);
        console.log(`Assets in ${assetDest}: ${files.length} items`);
    } else {
        console.log("Asset dest not found");
    }
} catch (e) {
    console.log("Error checking assets", e);
}
